backend/src/core/middlewares/auth.middleware.ts

```diff
@@ -128,19 +129,23 @@ export const InitAuthMiddleware = (authService: AuthService) => {
         action: 'authenticators[AuthType.Cookie]',
         user: req.session?.user,
       })
+      if (req.session) {
+        // default user to 10 requests per second if they're using cookie
+        req.session.rateLimit = DEFAULT_TX_EMAIL_RATE_LIMIT
+      }
       return true
     },
     [AuthType.ApiKey]: async (req: Request) => {
       const user = await authService.getUserForApiKey(req)
       if (user === null || !req.session) {
         return false
       }
-
       // Ideally, we store the user id in res.locals for api key, because theoretically, no session was created.
       // Practically, we have to check multiple places for the user id when we want to retrieve the id
       // To avoid these checks, we assign the user id to the session property instead so that downstream middlewares can use it
       req.session.user = user
       req.session.apiKey = true
+      req.session.rateLimit = user.rateLimit
```

Comment:
this ensures for any API call (by cookie or by API key), the `rateLimit` field will be set (making its subsequent use easier)